src/textual/css/stylesheet.py

```diff
@@ -179,47 +180,79 @@ def apply(self, node: DOMNode, animate: bool = False) -> None:
             },
         )
 
-        self.apply_rules(node, node_rules, animate=animate)
+        self.replace_rules(node, node_rules, animate=animate)
 
     @classmethod
-    def apply_rules(cls, node: DOMNode, rules: RulesMap, animate: bool = False) -> None:
-        """Apply style rules to a node, animating as required.
+    def replace_rules(
+        cls, node: DOMNode, rules: RulesMap, animate: bool = False
+    ) -> None:
+        """Replace style rules to a node, animating as required.
```

Comment:
This code was tricky to get right. There are a few things going on here. Let me know if you want to go over it.